### PR TITLE
Fixing #24214

### LINKED ISSUE #24214: no method matching isvalid(::SubString{String})
Although the docs say "Char" or "String"

```
Returns true if the given value is valid for that type. Types currently can be either Char or String.
```

SubStrings should be accepted as well, but they aren't.

```
isvalid(lstrip(" blabla"))
ERROR: MethodError: no method matching isvalid(::SubString{String})
Closest candidates are:
  isvalid(::SubString, ::Integer) at strings/types.jl:70
  isvalid(::AbstractString, ::Integer) at strings/basic.jl:228
  isvalid(::Char) at strings/utf8proc.jl:56
```

### PR DESCRIPTION
**Pull request by Vaibhavdixit02**
I did a PR previously for this but after #24999 I felt that it would be better to do the changes again because some of the files changed in the previous PR were altered.

Fixes #24214.

### CODE REVIEW

**Review by StefanKarpinski — COMMENTED** [review]
Looks good, how about adding a test?

**Comment by Vaibhavdixit02** [comment]
Added a couple of tests.

**Comment by StefanKarpinski** [comment]
There seems to be a whitespace error here, but it doesn't look like it was introduced by this PR.

File: test/strings/basic.jl [hunk 1]
```diff
@@ -428,6 +428,9 @@ end
     end
     # Check seven-byte sequences, should be invalid
     @test isvalid(String, UInt8[0xfe, 0x80, 0x80, 0x80, 0x80, 0x80]) == false
+    # issue 24214, Check valid SubString
+    @test isvalid(SubString("teststring",1,5)) == true
+    @test isvalid(SubString(String(UInt8[0xfe, 0x80, 0x80, 0x80, 0x80, 0x80]), 1,2)) == false
```
Comment (ararslan) [inline test/strings/basic.jl]: There's trailing whitespace on the end of this line

**Comment by Vaibhavdixit02** [comment]
I did not pay attention to it earlier, sorry!

**Pull request closed without merging**